emctl/cmd/client/command/meshinstall/coredns/deploy.go

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2021, MegaEase
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package coredns
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/megaease/easemeshctl/cmd/client/command/flags"
+	installbase "github.com/megaease/easemeshctl/cmd/client/command/meshinstall/base"
+
+	"github.com/pkg/errors"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	coreDNSNamespace   = "kube-system"
+	coreDNSDeployment  = "coredns"
+	coreDNSConfigMap   = "coredns"
+	coreDNSClusterRole = "system:coredns"
+)
+
+// Deploy deploy resources of coreDNS.
+func Deploy(ctx *installbase.StageContext) error {
+	err := installbase.BatchDeployResources(ctx,
+		[]installbase.InstallFunc{
+			configMapSpec(ctx),
+			clusterRoleSpec(ctx),
+
+			coreDNSDeploymentSpec(ctx),
+		})
+	if err != nil {
+		return err
+	}
+
+	return checkCoreDNSStatus(ctx.Client, ctx.Flags)
+}
+
+// PreCheck check prerequisite for installing mesh coreDNS
+func PreCheck(context *installbase.StageContext) error {
+	// Do nothing
+	return nil
+}
+
+// Clear clears all k8s resources about coreDNS
```

Comment:
We don't support revert installation of the coredns. Clear func should do nothing

Reply:
Updated